//source $JDE_DIR/Framework/common.sh
//cd $JDE_DIR/Public/jde/markets/types/proto;protoc --cpp_out dllexport_decl=JDE_MARKETS_EXPORT:. edgar.proto;sed -i -e 's/JDE_MARKETS_EXPORT/ΓM/g' edgar.pb.h;sed -i '1s/^/\xef\xbb\xbf/' edgar.pb.h;
//cd $JDE_DIR/MarketLibrary/source/types/proto;mv $JDE_DIR/Public/jde/markets/types/proto/edgar.pb.cc .;mklink edgar.pb.h $JDE_DIR/Public/jde/markets/types/proto
syntax = "proto3";
package Jde.Markets.Edgar.Proto;

message InfoTable
{
	string cusip=1;
	uint32 class_id=2;
	int32 value=3;
	int32 shares=4;
	bool is_principal=5;
}

message IndexLine
{
	uint32 cik=1;
	uint32 quarter=2;
	uint32 year=3;
	uint32 line_number=4;
}

message Form13FOld
{
	uint32 cik=1;
	uint32 quarter=2;
	uint32 year=3;
	uint32 line_number=4;
	int64 value=5;
	repeated InfoTable holdings=6;
}

message Form13F
{
	IndexLine line=1;
	int64 value=2;
	repeated InfoTable holdings=3;
}

message Form13G
{
	IndexLine line=1;
	int32 shares=5;
	string name=6;
}

message File13F
{
	repeated Form13F forms=1;
}

message File13FOld
{
	repeated Form13FOld forms=1;
}

message File13G
{
	repeated Form13G forms=1;
}

message Investment
{
	InfoTable table=1;
	float percent=2;
}

message Investor
{
	uint32 cik=1;
	uint32 end_period=2;
	repeated Investment current=3;
	repeated Investment previous=4;
}
message Company
{
	uint32 cik=1;
	string name=2;
}
message Investors
{
	uint32 request_id=1;
	repeated Investor values=2;
	repeated Company companies=3;
}

message Filing
{
	uint32 line_number=1;
	uint32 cik=2;
	uint32 form_id=3;
	uint32 submitter=4;
	uint32 year=5;
	uint32 sequence=6;
	uint32 filing_day=7;
	uint32 period_end=8;
}

message Filings
{
	uint32 request_id=1;
	repeated Filing values=2;
}

message MasterIndex
{
	uint32 quarter=1;
	uint32 year=2;
	repeated Filing filings=3;
}
